lib/lints/banned_external_code/models/banned_external_code_entry_parameters.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:analyzer/error/error.dart';
+import 'package:solid_lints/utils/parameter_utils.dart';
+
+/// A data model class that represents the "entries" input
+/// parameters. All parameters are optional, which allows global variables,
+/// classes, and entire packages to be linted.
+///
+/// Parameters:
+/// * id: Variable/method name
+/// * class_name: Name of the class containing the variable/method
+/// * source: Package (e.g., dart:async or package:example)
+/// * severity: The default severity of the lint for each entry.
+/// * reason: Explain why the code is banned
+/// * includes: Glob patterns for files to include
+/// * excludes: Glob patterns for files to exclude
+///
+/// Example:
+/// ```yaml
+/// entries:
+///   - id: wait
+///     class_name: Future
+///     source: dart:async
+///     reason: "Future.wait from dart:async isnt allowed"
+///     severity: warning
+///     includes: []
+///     excludes: []
+/// ```
+class BannedExternalCodeEntryParameters {
+  /// Variable/method name
+  final String? id;
+
+  /// Name of the class containing the variable/method
+  final String? className;
+
+  /// Package (e.g., dart:async or package:example)
+  final String? source;
```

Comment:
IMO `source` should be required, and limited to external packages.

Whatever code is in the current package -- we can safely assume we have control of, and there already is a `@Deprecated` mechanism in place for cases where we want to phase out the usage of some internal API.

I'd rather not introduce an alternative way to do the same thing.

Reply:
> I agree, though we can't stop users from specifying the source as package:my_app.

Well, if the users want to use a hacky loophole -- who are we to stop their creativity :)